src/test/java/spoon/test/enums/EnumsTest.java

```diff
@@ -16,136 +16,139 @@
-				+ "}", burritos.toString());
+		assertThat(burritos.prettyprint(), containsRegexMatch("Burritos \\{\\W+;"));
 	}
 
 	@Test
-	public void testGetAllMethods() throws Exception {
+	void testGetAllMethods() throws Exception {
 		// contract: getAllMethods also returns the methods of Enum
 		final Factory factory = build(Burritos.class);
 		final CtType<Burritos> burritos = factory.Type().get(Burritos.class);
-		CtMethod name = factory.Core().createMethod();
+		CtMethod<String> name = factory.Core().createMethod();
 		name.setSimpleName("name"); // from Enum
 		name.setType(factory.Type().createReference(String.class));
 		assertTrue(burritos.hasMethod(name));
 		assertTrue(burritos.getAllMethods().contains(name));
+		// this does not work due to the SignatureBasedSortedSet violating the Set contract
+		// assertThat(burritos.getAllMethods(), hasItem(name));
 	}
 
-	@Test
-	public void testNestedPrivateEnumValues() throws Exception {
+	@ParameterizedTest
+	@ArgumentsSource(NestedEnumTypeProvider.class)
+	void testEnumValueModifiers(CtEnum<?> type, CtExtendedModifier visibility) {
 		// contract: enum values have correct modifiers
-		CtType<?> ctClass = ModelUtils.buildClass(NestedEnums.class);
-		{
-			CtEnum<?> ctEnum = ctClass.getNestedType("PrivateENUM");
-			assertEquals(asSet(ModifierKind.PRIVATE), ctEnum.getModifiers());
-			assertEquals(asSet(ModifierKind.PRIVATE, ModifierKind.STATIC, ModifierKind.FINAL), ctEnum.getField("VALUE").getModifiers());
-		}
-		{
-			CtEnum<?> ctEnum = ctClass.getNestedType("PublicENUM");
-			assertEquals(asSet(ModifierKind.PUBLIC), ctEnum.getModifiers());
-			assertEquals(asSet(ModifierKind.PUBLIC, ModifierKind.STATIC, ModifierKind.FINAL), ctEnum.getField("VALUE").getModifiers());
-		}
-		{
-			CtEnum<?> ctEnum = ctClass.getNestedType("ProtectedENUM");
-			assertEquals(asSet(ModifierKind.PROTECTED), ctEnum.getModifiers());
-			assertEquals(asSet(ModifierKind.PROTECTED, ModifierKind.STATIC, ModifierKind.FINAL), ctEnum.getField("VALUE").getModifiers());
-		}
-		{
-			CtEnum<?> ctEnum = ctClass.getNestedType("PackageProtectedENUM");
-			assertEquals(asSet(), ctEnum.getModifiers());
-			assertEquals(asSet(ModifierKind.STATIC, ModifierKind.FINAL), ctEnum.getField("VALUE").getModifiers());
+		if (visibility != null) {
+			assertThat(type.getField("VALUE").getExtendedModifiers(), contentEquals(
+					// TODO this is wrong, the field should be public
+					new CtExtendedModifier(visibility.getKind(), true),
+					new CtExtendedModifier(ModifierKind.STATIC, true),
+					new CtExtendedModifier(ModifierKind.FINAL, true)
+			));
+		} else {
+			assertThat(type.getField("VALUE").getExtendedModifiers(), contentEquals(
+					// TODO this is wrong, the field should be public (and then the if else,
+					//  the visibility param and the Streams#zip can be removed)
+					new CtExtendedModifier(ModifierKind.STATIC, true),
+					new CtExtendedModifier(ModifierKind.FINAL, true)
+			));
```

Comment:
I don't think this is an improvement over the original test. While the original has the problem that it's 4 tests in one, it's very easy to read. The refactored version has 2 mutually exclusive branching paths (in general, avoid branches in tests) and also requires one to understand the non-trivial parameterization.

I favor the original test, it's much easier to grasp. I'm guessing you have a plan here for when you fix the modifier bug, but that's not a concern of this PR. Could you revert to the original? Feel free to reintroduce the parameterized test if it actually simplifies things in a future PR.